pyapr/converter/converter_methods.py

```diff
@@ -1,138 +1,168 @@
 import pyapr
+from pyapr.converter import FloatConverter, ShortConverter
 import numpy as np
+from typing import Union, Optional
+
+
+__allowed_image_types__ = [np.uint8, np.uint16, np.float32]
+
+
+def get_apr(image: np.ndarray,
+            converter: Optional[Union[FloatConverter, ShortConverter]] = None,
+            params: Optional[pyapr.APRParameters] = None,
+            verbose: bool = False):
+    """
+    Convert an image array to the APR.
+
+    Parameters
+    ----------
+    image: numpy.ndarray
+        Input (pixel) image as an array of 1-3 dimensions.
+    converter: FloatConverter or ShortConverter (optional)
+        Converter object used to compute the APR. By default, FloatConverter is used to avoid rounding errors in
+        internal steps.
+    params: pyapr.APRParameters (optional)
+        If provided, sets parameters of the converter. If not, the parameters of the converter object is used, with
+        'auto_parameters' set to True.
+    verbose: bool
+        Set the verbose mode of the converter (default: False).
+
+    Returns
+    -------
+    A tuple (apr, parts) where apr is the APR object and parts is the sampled particle data object of type
+    FloatParticles or ShortParticles (depending on the input image data type).
+    """
+
+    if image.dtype not in __allowed_image_types__:
+        errstr = 'pyapr.converter.get_apr accepts images of type float32, uint16 and uint8, ' \
+                 'but {} was given'.format(image.dtype)
+        raise TypeError(errstr)
 
-
-def get_apr(image, rel_error=0.1, gradient_smoothing=2, verbose=True, params=None):
-
-    # check that the image array is c-contiguous
-    if not image.flags['C_CONTIGUOUS']:
-        print('WARNING: \'image\' argument given to get_apr is not C-contiguous \n'
-              'input image has been replaced with a C-contiguous copy of itself')
-        image = np.ascontiguousarray(image)
```

Comment:
Is this no longer required?

Reply:
I added this on the C++ end, it should throw if you give in something else. I'm not sure how much info you get though, I'll try to break it and see, and add this back in if the error is unclear

Reply:
I added this back in - the error returned did not say anything about data layout